Add example successful documentation bugs (#188)
r-devel/rdevguide: 08-documenting.Rmd

## Helping with documentation

Maintaining the accuracy of R's documentations and keeping a high level of quality takes a lot of effort. Community members, like you, help with writing, editing, and updating content, and these contributions are appreciated and welcomed.

Looking at pre-existing documentation source files can be very helpful when getting started.

You can directly search for [documentation issues/bugs on Bugzilla](https://bugs.r-project.org/buglist.cgi?component=Documentation&list_id=22501&product=R&resolution=---). Issues vary from typos to unclear documentation and items lacking documentation.

If you see a documentation issue that you would like to tackle, you can leave a comment on the issue saying you are going to try to solve the issue and mention roughly how long you think you will take to do so (this allows others to take on the issue if you happen to forget or lose interest).

## Reporting documentation bugs
 <!--
discussion in slack (random channel)
-->

> If you find some typo or problem on the [CRAN](https://cran.r-project.org) webpages you can write a polite email to `CRAN@r-project.org` to report it. As an alternative, one can also write to the R-devel mailing list or submit a bug report via R's Bugzilla.

Reviewer: This belongs elsewhere (as you suggest in #184), since the CRAN website is not part of (base) R's documentation, so we can delete this. 
```suggestion
```

> For improvements of the R-manuals or reporting typos or bugs in the R-manuals, submit a bug report to [R's Bugzilla](https://bugs.r-project.org/).

Reviewer: This also applies to documentation issues in the base R packages, or other packages maintained by the R Core Team. Here we can link to https://contributor.r-project.org/rdevguide/ReportingBugs.html#ReportBug.

Author: I've added the link to reporting bugs

> For reporting bugs or typos on the webpages or documentation about a particular package, write to the corresponding package maintainer. To find the maintainer of a package, use the command `maintainer("package-name")`.

Reviewer: Need to differentiate between packages maintained by R core and contributed packages. See https://contributor.r-project.org/rdevguide/ReportingBugs.html#ReportBug.

Author: Have tried to make this section a bit clearer

Reviewer: Thanks, made a couple of suggestions to finalise this part.

## Helping with the Developer's Guide

> The Developer's Guide (what you are reading now) uses the same process as the main R documentation, except for some small differences. The source lives in a [GitHub repository](https://github.com/r-devel/rdevguide/) and bug reports should be submitted to the [devguide GitHub tracker](https://github.com/r-devel/rdevguide/issues).

Reviewer: ```suggestion
Contributions to the R Development Guide (what you are reading now) are also welcome. The source for this guide lives in a [GitHub repository](https://github.com/r-devel/rdevguide/) and bug reports should be submitted to the [issue tracker](https://github.com/r-devel/rdevguide/issues).
```

> Our Developer's Guide workflow uses continuous integration and deployment so changes to the Developer’s Guide are normally published when the pull request is merged. [How to contribute to this guide from the introduction.](#how-to-contribute-to-this-guide)

Reviewer: Either briefly explain what a pull request is, or briefly explain that people can also propose changes to the source directly as described in the linked section.

Author: Shuffled parts around, pointing to the introduction section first, then adding a bit of context about PRs before the final paragraph about CI - how does that read now?

Reviewer: Looks good - just suggested a minor change to be consistent about the name of the guide.
